appium/lib/screen-object.rb

```diff
@@ -69,82 +69,97 @@ def enter
     scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
+    x = scr['width'] / 4
+    y = scr['height'] / 2.8
     for i in 0..num_loop
       begin
-        if (driver.find_element(locator,locator_value).displayed?)
+        if (driver.find_element(locator).displayed?)
           break
         end
       rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
+        scroll(direction)
+        sleep post_timeout
         false
       end
+      raise("#{locator_value} is not displayed") if i==num_loop
     end
   end
 
-  def scroll_down_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
+  def scroll_down_click(locator, post_timeout = 2, num_loop = 15)
+    size = driver.driver.manage.window.size
+    height = size.height
+    width = size.width
     for i in 0..num_loop
       begin
         if (driver.find_element(locator,locator_value).displayed?)
           driver.find_element(locator,locator_value).click
           break
         end
       rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
+        scroll(:down)
+        sleep post_timeout
         false
       end
+      raise("#{locator.locator} is not displayed") if i==num_loop
     end
   end
 
-  def scroll_up_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
+  def scroll_up_find(locator, post_timeout = 2, num_loop = 15)
     for i in 0..num_loop
       begin
         if (driver.find_element(locator,locator_value).displayed?)
           break
         end
       rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
+        scroll(:up)
+        sleep post_timeout
         false
       end
+      raise("#{locator.locator} is not displayed") if i==num_loop
     end
   end
 
-
-  def scroll_up_click(locator,locator_value,num_loop = 15)
+  def scroll_up_click(locator, post_timeout = 2, num_loop = 15)
     scr = driver.window_size
     screenHeightStart = (scr.height) * 0.5
     scrollStart = screenHeightStart.to_i
     screenHeightEnd = (scr.height) * 0.2
     scrollEnd = screenHeightEnd.to_i
     for i in 0..num_loop
       begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
+        if (driver.find_element(locator).displayed?)
```

Comment:
This if statement can easily be a one-liner since it's a guard clause I think it would be appropriate.